src/js/components/FileInput/FileInput.js

```diff
@@ -130,133 +130,206 @@ const FileInput = forwardRef(
-                {files.length} {messages.files || 'files'}
-              </Label>
+        {(!files.length || files.length > 1) && (
+          <Box direction="row" justify="between">
+            <Message {...theme.fileInput.message}>{message}</Message>
+            <Keyboard
+              onSpace={event => {
+                if (controlRef.current === event.target)
+                  inputRef.current.click();
+              }}
+              onEnter={event => {
+                if (controlRef.current === event.target)
+                  inputRef.current.click();
+              }}
+            >
+              {theme.fileInput.button ? (
+                <Button
+                  ref={controlRef}
+                  kind={theme.fileInput.button}
+                  label={messages.browse || 'browse'}
+                  onClick={() => {
+                    inputRef.current.click();
+                    inputRef.current.focus();
+                  }}
+                  {...rest}
+                />
+              ) : (
+                <Anchor
+                  alignSelf="center"
+                  ref={controlRef}
+                  margin="small"
+                  onClick={() => {
+                    inputRef.current.click();
+                    inputRef.current.focus();
+                  }}
+                  tabIndex={0}
+                  label={messages.browse || 'browse'}
+                />
+              )}
+            </Keyboard>
+          </Box>
+        )}
+        {files.length > aggregateThreshold && (
+          <Box direction="row" align="center" justify="between">
+            <Label {...theme.fileInput.label}>
+              {files.length} {messages.files || 'files'}
+            </Label>
+            <Button
+              ref={removeRef}
+              a11yTitle={messages.removeAll || 'remove all'}
+              icon={<RemoveIcon />}
+              hoverIndicator
+              onClick={event => {
+                event.stopPropagation();
+                setFiles([]);
+                inputRef.current.focus();
+              }}
+            />
+          </Box>
+        )}
+        {files.length > 0 &&
+          files.length <= aggregateThreshold &&
+          files.map((file, index) => (
+            <Box
+              key={file.name}
+              direction="row"
+              align="center"
+              justify="between"
+            >
+              {renderFile ? (
+                renderFile(file)
+              ) : (
+                <Label
+                  weight={
+                    theme.global.input.weight || theme.global.input.font.weight
+                  }
+                  truncate
+                  {...theme.fileInput.label}
+                >
+                  {file.name}
+                </Label>
+              )}
               <Button
-                ref={removeRef}
-                a11yTitle={messages.removeAll || 'remove all'}
+                ref={index ? undefined : removeRef}
```

Comment:
I can't click on the remove button, I can only tab to it. Also, I feel like the spacing here is a bit awkward. I would expect the X to be closer to the browse
<img width="445" alt="Screen Shot 2021-04-26 at 11 24 19 AM" src="https://user-images.githubusercontent.com/12522275/116134160-72644580-a684-11eb-970e-e6a7f4e3c2a2.png">

Reply:
Ill look into the behavior I bet the `input` is covering the `x` my math may be off...
But Yes I just looked at figma and it should be closer to the `browse` button thanks Ill fix!

Reply:
@taysea which story was this ? that you were not able to click on the `X`?

Reply:
This was on "Simple"

Reply:
@taysea check now I just updated the calculation 👍

Reply:
Perfect, looks great now!

Reply:
Noticing something else now though, if I add a file with a long filename, the X and browse are being allowed to expand outside of the fileinput:

<img width="502" alt="Screen Shot 2021-04-26 at 2 50 56 PM" src="https://user-images.githubusercontent.com/12522275/116155423-ead80000-a69e-11eb-94ee-4a36f8e86c10.png">

Reply:
oh no :( okay ill look into that!